requirements.txt

```diff
@@ -0,0 +1,11 @@
+python
+
+numpy
+ipopt
+cython
+setuptools
+pkg-config
+six
+future
+
+pytest
```

Comment:
Maybe name this `dev-requirements.txt` since it has pytest.

Reply:
Agreed. If we end up moving the tests to `cyipopt/tests/` then I'll move to `dev-requirements.txt` but if we leave `tests/` at the root then I'll make a separate `tests/requirements.txt` (as well as a similar thing for the documentation requirements in `docs/requirements.txt`)? Or would you prefer all to just be in `dev-requirements.txt` whatever?

Reply:
What is the use case of having the requirements.txt file? I rarely use them, as I've switched to conda based development environments years ago.

Reply:
I personally find it useful to have so I can run `conda install --file requirements.txt` to easily make a new conda-env with the correct requirements if I'm developing. Similarly, as part of CI, creating the virtual environments with the minimum requirements is good for checking everything installs correctly. I tend to only install the test requirements to the virtual environment in CI just before running the test suite and the same for building the docs. 

Again, this may be unnecessary and a bad habit I've picked up so happy to follow a different approach here if there's a better one.

Reply:
Ah, got it. Never thought to use it that way with conda. If this is mostly for developers, then you can leave it however you see fit. I have no objections.

Reply:
Addressed in https://github.com/matthias-k/cyipopt/pull/75/commits/9302a7e9a2ce9a1fde4baeb47c2c65504b153681, https://github.com/matthias-k/cyipopt/pull/75/commits/39833ed1365c4bef2cef9e8603b387c8912d8233, https://github.com/matthias-k/cyipopt/pull/75/commits/c260e1ea0491ec39de3fa71482a58474400f9de8 and https://github.com/matthias-k/cyipopt/pull/75/commits/62b85d56d1305d491fbace1ecd603105afa0d8c0.